src/checkers/inference/solver/backend/encoder/ConstraintEncoderCoordinator.java

```diff
@@ -73,6 +74,34 @@
         }
     }
 
+    public static <ConstraintEncodingT> ConstraintEncodingT dispatch(
+            ArithmeticConstraint constraint,
+            ArithmeticConstraintEncoder<ConstraintEncodingT> encoder) {
+        switch (SlotSlotCombo.valueOf(constraint)) {
+            case VARIABLE_VARIABLE:
+                return encoder.encodeVariable_Variable(constraint.getOperation(),
+                        (VariableSlot) constraint.getLeftOperand(),
+                        (VariableSlot) constraint.getRightOperand(),
+                        (VariableSlot) constraint.getResult());
+            case VARIABLE_CONSTANT:
+                return encoder.encodeVariable_Variable(constraint.getOperation(),
+                        (VariableSlot) constraint.getLeftOperand(),
+                        (ConstantSlot) constraint.getRightOperand(),
+                        (VariableSlot) constraint.getResult());
+            case CONSTANT_VARIABLE:
+                return encoder.encodeVariable_Variable(constraint.getOperation(),
+                        (ConstantSlot) constraint.getLeftOperand(),
+                        (VariableSlot) constraint.getRightOperand(),
+                        (VariableSlot) constraint.getResult());
+            case CONSTANT_CONSTANT:
+                return encoder.encodeVariable_Variable(constraint.getOperation(),
```

Comment:
If you using the same `encodeVariable_Variable()` method, what is the need of different type casting on the parameters in each cases? Why not centralize these cases to flow into the same block? e.g.

```java
case V_V:
case V_C:
case C_V:
case C_C:
{
 // share the code, as they are actually same except the type castings (but why type castings at all).
}
```

Reply:
woops, should be calling each V/C combo in the encoder. Thanks for the catch (although the logic remains the same in the encoder I have haha).